actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -94,13 +94,17 @@ def authenticate(request, &login_procedure)
       end
 
       def has_basic_credentials?(request)
-        request.authorization.present? && (auth_scheme(request).downcase == 'basic')
+        request.authorization.present? && (auth_scheme(request).downcase == 'basic') && !user_name_or_password_blank?(request)
       end
 
       def user_name_and_password(request)
         decode_credentials(request).split(':', 2)
       end
 
+      def user_name_or_password_blank?(request)
```

Comment:
@gregmolnar Ooof, De Morgan's law bites me again. :)

I'm not convinced that this is a framework-level concern. If someone *wants* to deliberately use an empty password for Basic auth it may be questionable security - but it's not incorrect according to RFC2617.